src/javascript/JContent/AdditionalAppsTree/AdditionalAppsTree.jsx

```diff
@@ -1,26 +1,44 @@
-import React from 'react';
+import React, {useEffect} from 'react';
 import {TreeView} from '@jahia/moonstone';
 import {useDispatch, useSelector} from 'react-redux';
 import {cmGoto} from '../JContent.redux';
 import {useAdminRouteTreeStructure} from '@jahia/ui-extender';
-import {useHistory} from 'react-router';
 import {useNodeInfo} from '@jahia/data-helper';
 import {useTranslation} from 'react-i18next';
 import PropTypes from 'prop-types';
 
 export const AdditionalAppsTree = ({item, target}) => {
-    const site = useSelector(state => state.site);
+    const {site, path} = useSelector(state => ({site: state.site, path: state.jcontent.path}));
     const dispatch = useDispatch();
     const {t} = useTranslation();
-    const history = useHistory();
 
-    let selected = history.location.pathname.split('/').pop();
+    const selected = path.substr(1);
 
-    const {tree, defaultOpenedItems, allPermissions} = useAdminRouteTreeStructure(target, selected);
+    const {tree, routes, defaultOpenedItems, allPermissions} = useAdminRouteTreeStructure(target, selected);
 
-    const {node, loading, error} = useNodeInfo({path: '/sites/' + site}, {getPermissions: allPermissions, getSiteInstalledModules: true});
+    const {node, loading, error} = useNodeInfo({path: '/sites/' + site}, {
+        getPermissions: allPermissions,
+        getSiteInstalledModules: true
+    });
 
-    if (loading || error) {
+    let switchSelection;
+    if (!loading && !error && selected === '') {
+        const firstItem = routes.find(route => route.isSelectable &&
+            (route.requiredPermission === undefined || node[route.requiredPermission] !== false) &&
+            (route.requireModuleInstalledOnSite === undefined || node.site.installedModulesWithAllDependencies.indexOf(route.requireModuleInstalledOnSite) > -1)
+        );
+        if (firstItem) {
+            switchSelection = firstItem.key;
+        }
+    }
+
+    useEffect(() => {
+        if (switchSelection) {
+            dispatch(cmGoto({path: '/' + switchSelection}));
+        }
+    }, [dispatch, switchSelection]);
```

Comment:
What is the reason for passing 'dispatch' here ? I have read https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects and do understand the reason for passing 'switchSelection', but for 'dispatch' I don't get it. Maybe I am missing something...

Reply:
It's quite useless, but since dispatch is used in the body of the callback , it must be part of the dependencies. Removing it won't get through the linter (which will add it automatically, btw)